updated palette to include extended and additional components

jonborchardt opened the pull request:
added extended palette
added styles for a few components
updated demos to look a bit better
updated footer to use palette/theme

usage of color.X or color2.X in components should be limited, rather we should use palette and extended

jonborchardt commented on docs/data/material/varnish/colors/colors.md:
```diff
@@ -29,22 +29,22 @@ e.g. `color.B6` -> `color2.B3`, `color.R9` -> `color2.R5`.
 eg: `theme.color2.B3`
 
-{{"demo": "DefaultDemo.js", "bg": false, "defaultCodeOpen":false}}
+{{"demo": "DefaultDemo.js", "bg": false, "defaultCodeOpen": false}}
```
just making the demos look cleaner

jonborchardt commented on docs/data/material/varnish/varnish-app/Extending.js:
```diff
@@ -14,7 +14,7 @@ export default function Demo() {
 
   const CoolH1 = styled.h1`
-    color2: ${({ theme }) => theme.color2.Freeze};
+    color: ${({ theme }) => theme.color2.Freeze};
```
oops

jonborchardt commented on docs/next.config.js:
```diff
@@ -23,7 +23,7 @@ module.exports = withDocsInfra({
       {
         source: '/',
-        destination: '/material-ui/getting-started/overview/',
+        destination: '/material-ui/varnish/varnish-app/',
```
better landing for now

jonborchardt commented on packages/varnish/src/components/Footer.tsx:
```diff
@@ -15,38 +15,33 @@ interface Props {
-const StyledFooter = styled(({ textColor, background, ...rest }) => <footer {...rest} />)<{
-  textColor: string;
-  background: string;
+const StyledFooter = styled(({ backgroundColor, isContrast, ...rest }) => <footer {...rest} />)<{
```
making footer use theme instead of hard coded colors

jonborchardt commented on packages/varnish/src/components/header/AI2Banner.tsx:
```diff
@@ -19,7 +19,7 @@ const BannerContent = styled.div`
 
 const DarkBanner = styled.div`
-  background: ${({ theme }) => theme.color2.B5.toString()};
+  background: ${({ theme }) => theme.extended.background.dark};
```
we should use palette and extended more

arnavic reviewed (approved):
How should we consider things going into extended vs. palette? As far as new colors etc.?

jonborchardt commented:
> How should we consider things going into extended vs. palette? As far as new colors etc.?

mui has a theme that has a palettet object.
```
theme = {
  palette: {...}
}
```

this object should only contain properties as defined by mui
and we should use those theme properties whenever it makes sense... i.e. when we want the text color, we should use: `theme.palette.text.primary` (vs. `theme.color.N5`)

BUT when we want to have new palette values, ones not expressed in the theme.palette, and stuff we use more then once... and or think others may want to style... we add to `extended` <- im going to rename this to `extendedPalette`)
e.g. we use a dark background so i added a section called background and added dark to it...
we can add more as needed...
